pandas/tests/groupby/test_function.py

```diff
@@ -892,6 +892,14 @@ def test_nunique_with_timegrouper():
     tm.assert_series_equal(result, expected)
 
 
+def test_nunique_preserves_column_level_names():
+    # GH 23222
+    test = pd.DataFrame([1, 2, 3],
+                        columns=pd.Index(['A'], name="level_0"))
+    result = test.groupby([0, 0, 0]).nunique()
+    tm.assert_index_equal(test.columns, result.columns)
```

Comment:
I know the bug is focused on the names but for robustness, let's compare the entire `DataFrame`.

Reply:
Heh.  I attended a talk on the weekend where the speaker argued for single-purpose tests and against byproduct testing. ;-)  But it's easy to change, and I'm happy to respect house style.

Reply:
Well, consider your single purpose to check that the output of `nunique` is correct 😉.  Thanks!